thetis/solver2d.py

```diff
@@ -299,7 +299,7 @@ def create_equations(self):
         self.eq_sw.bnd_functions = self.bnd_functions['shallow_water']
         if self.options.solve_tracer:
             self.fields.tracer_2d = Function(self.function_spaces.Q_2d, name='tracer_2d')
-            if self.options.timestepper_type == 'CrankNicolson':
+            if self.options.timestepper_type in ('CrankNicolson', 'SSPRK33', 'ForwardEuler', 'BackwardEuler', 'DIRK22', 'DIRK33', 'CrankEuler'):
```

Comment:
In my opinion, testing for `timestepper_type` does not belong to to `create_equations` function, would fit more naturally in `create_timestepper`. Just create the equation if `solve_tracer` is set.